File: python/functions/anonymous_functions.py
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
# SPDX-License-Identifier: GPL-3.0-or-later

simple_function = lambda : print("Simple function.")

function_returning_value = lambda : 9

function_with_arguments = lambda number, text : print(f"A function with some arguments:\nnumber: {number}\ntext: {text}")

i = 10

print("Functions:\n")

simple_function()
print()

i = function_returning_value()
print(f"returned value: {i}\n")

function_with_arguments(6, "orange")
print()

File: c++/functions/anonymous_functions.cpp
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */
// SPDX-License-Identifier: GPL-3.0-or-later

#include <iostream>

int main()
{
  auto simple_function = [] () -> void { std::cout << "Simple function." << std::endl; };

  auto function_with_local_variable = [] () -> void
  {
    int i = 4;
    std::cout << "A function with a local variable: " << i << std::endl;
  };

  auto function_returning_value = [] () -> int
  {
    std::cout << "A function returning value." << std::endl;
    return 9;
  };

  auto function_with_arguments = [] (int number, const char* text) -> void
  {
    std::cout << "A function with some arguments:\n"
      << "number: " << number
      << std::endl
      << "text: " << text
      << std::endl;
  };

  int i = 10;

  std::cout << "Functions:\n\n";

  simple_function();
  std::cout << std::endl;

  function_with_local_variable();
  std::cout << std::endl;

  i = function_returning_value();
  std::cout << "returned value: " << i << std::endl << std::endl;

  function_with_arguments(6, "orange");
  std::cout << std::endl;
}
